### PR TITLE
Vam 42/addressable slideshow

### PR DESCRIPTION
**Pull request by ionab**
This PR makes each slide in a slideshow addressable.

### CODE REVIEW

File: src/canvasPanelComponents/slideshow/src/components/CanvasNavigation/CanvasNavigation.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+        this.goToSlide(0);
+        return;
+      } else {
+        this.props.goToRange(parseInt(slideId));
+      }
+    }
+  }
+  goToSlide = index => {
+    if (this.props.addressable) {
+      document.location.hash = '#?slide=' + index;
```
Comment (stephenwf) [inline src/canvasPanelComponents/slideshow/src/components/CanvasNavigation/CanvasNavigation.js]: Do you think we could add one more prop to the base component (like `id="1234"`) that could also be checked against the hash? (`#?id=1234&slide=1234`) I think it's a common case to have one or more of these on the page.

**Comment by ionab** [comment]
Note: Storybook manipulates the url to allow the user to return to a direct story: 
You can however view the url changes in the iframe mode: 
https://5e5fe3e49757e70009f31921--romantic-wescoff-9fe96d.netlify.com/iframe.html?id=slideshow-slideshow--slideshow-story#?slide=3

File: src/canvasPanelComponents/slideshow/src/components/CanvasNavigation/CanvasNavigation.js [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
 import { withBemClass } from '@canvas-panel/core';
+import withLocation from '../withLocation/withLocation';
 
 import './CanvasNavigation.scss';
 
 class CanvasNavigation extends Component {
+  componentDidMount() {
+    if (this.props.addressable && this.props.hash) {
+      let slideId =
+        this.props.hash.includes('slide') && this.props.hash.split('=')[1];
```
Comment (stephenwf) [inline src/canvasPanelComponents/slideshow/src/components/CanvasNavigation/CanvasNavigation.js]: We could re-use `queryString.parse(this.props.hash)` like the `withLocation` component. It should return an object `{ slide: 1234 }`.

File: src/canvasPanelComponents/slideshow/src/components/CanvasNavigation/CanvasNavigation.js [hunk 3]
```diff
@@ -23,6 +47,7 @@ class CanvasNavigation extends Component {
           onClick={ev => {
             ev.preventDefault();
             previousRange();
+            this.goToSlide(currentIndex - 1);
```
Comment (stephenwf) [inline src/canvasPanelComponents/slideshow/src/components/CanvasNavigation/CanvasNavigation.js]: I think the `currentIndex` might be changed from outside of this component, so there's no guarantee that the slide will always be updated. We could use `componentWillReceiveProps` (or the modern eq. I forget), and set it when the `currentIndex ` prop changes.

**Pull request merged**